src/theory/arith/theory_arith.cpp

```diff
@@ -355,13 +355,13 @@ bool TheoryArith::collectModelInfo(TheoryModel* m,
 {
   // If we have a buffered lemma (from the non-linear extension), then we
   // do not assert model values, since those values are likely incorrect.
-  // Moreover, the model does not need to satisfy the assertions, so
-  // arbitrary values can be used for arithmetic terms. Hence, we do
-  // nothing here. The buffered lemmas will be sent immediately
-  // at LAST_CALL effort (see postCheck).
+  // We send the lemmas now, as the model construction will otherwise fail.
```

Comment:
Is this related to the bug I was investigating, or is this a separate fix?

Reply:
Based on the more pedantic behavior (i.e. never assign to evaluatable equivalence classes), this behavior of the NL solver is no longer allowed, as it would lead to a bogus model.

Previously, we had relied on this deadlock-breaking behavior to assign (incorrect) values to equivalence classes with the promise we'd send these lemmas at last call after model construction.  I think its best to just disallow this behavior altogether.  

Note that 2 benchmarks unfortunately time out, which is due to the relationship between when these buffer lemmas get sent (full vs. last call effort).